src/controller/CHIPDevice.cpp

```diff
@@ -171,14 +171,15 @@ CHIP_ERROR Device::Serialize(SerializedDevice & output)
 
     CHIP_ZERO_AT(serializable);
 
-    serializable.mOpsCreds   = mPairing;
-    serializable.mDeviceId   = Encoding::LittleEndian::HostSwap64(mDeviceId);
-    serializable.mDevicePort = Encoding::LittleEndian::HostSwap16(mDeviceUdpAddress.GetPort());
-    serializable.mAdminId    = Encoding::LittleEndian::HostSwap16(mAdminId);
-    SuccessOrExit(error = Inet::GetInterfaceName(mDeviceUdpAddress.GetInterface(), Uint8::to_char(serializable.mInterfaceName),
+    serializable.mOpsCreds        = mPairing;
+    serializable.mDeviceId        = Encoding::LittleEndian::HostSwap64(mDeviceId);
+    serializable.mDevicePort      = Encoding::LittleEndian::HostSwap16(mDeviceAddress.GetPort());
+    serializable.mAdminId         = Encoding::LittleEndian::HostSwap16(mAdminId);
+    serializable.mDeviceTransport = mDeviceAddress.GetTransportType();
+    SuccessOrExit(error = Inet::GetInterfaceName(mDeviceAddress.GetInterface(), Uint8::to_char(serializable.mInterfaceName),
                                                  sizeof(serializable.mInterfaceName)));
     static_assert(sizeof(serializable.mDeviceAddr) <= INET6_ADDRSTRLEN, "Size of device address must fit within INET6_ADDRSTRLEN");
-    mDeviceUdpAddress.GetIPAddress().ToString(Uint8::to_char(serializable.mDeviceAddr), sizeof(serializable.mDeviceAddr));
+    mDeviceAddress.GetIPAddress().ToString(Uint8::to_char(serializable.mDeviceAddr), sizeof(serializable.mDeviceAddr));
```

Comment:
If we plan to enforce endianness in serialize, we should do the counter-conversion here.

Reply:
I marked Transport::Type as uint8_t instead, and placed static_assert to gurantee that the underlying type of Transport::Type is uint8_t.